### PR TITLE
Laravel 5

### PR DESCRIPTION
**Pull request by dlg-mambo**
Hi,

I had to use Trucker on a project and since I started fresh, I made the changes for Laravel 5. You might see a lot of useless commits since I was just confused about how Laravel 5 worked for config setting and getting ++ combined with the ConfigManager.php.

Everything seems to work fine for now. Since it's not a refactoring job, I left Guzzle to its original version in composer.json, but I guess someone who worked on the project could just verify if the latest version could make it.

Also, I changed only one line in the readme.md to reflect the new Laravel 5 way to publish vendor configs, but feel free to review if anything else is missing!

### CODE REVIEW

File: src/Trucker/TruckerServiceProvider.php [hunk 1]
```diff
@@ -202,25 +199,11 @@ public function bindClasses(Container $app)
...
+        $this->publishes([
+            __DIR__ . '/../config/trucker.php' => config_path('trucker.php'),
+        ]);
 
-                $customItems = include $customItems;
-
-                return array_replace_recursive($items, $customItems);
-            });
-        }
+        $this->mergeConfigFrom(__DIR__ . '/../config/trucker.php', 'trucker');
```
Comment (bkuhl) [inline src/Trucker/TruckerServiceProvider.php]: Have you verified that these config changes are functional?

File: composer.json [hunk 2]
```diff
@@ -24,13 +24,13 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
-        "illuminate/support": "~4",
-        "illuminate/config": "~4",
-        "illuminate/container": "~4",
+        "php": ">=5.5.9",
```
Comment (bkuhl) [inline composer.json]: Since this requirement is bumped, can you remove PHP 5.4 from the .travis.yml?

**Comment by bkuhl** [comment]
Thanks for digging into this!  Can you take a look at TravisCI and make the necessary corrections to get the build passing and then I'll re-review?

**Comment by david-treblig** [comment]
I did use it in production for a time until the project shifted to another solution and everything was working fine, but I don't have the setup to test things manually anymore.

**Comment by david-treblig** [comment]
I think the testes would have to be rewritten completely, but everything seemed to work on prod.

**Comment by bkuhl** [comment]
In my experience with L5 is that it's actually best to avoid writing tests for it since it uses raw functions now and usually relies on the IOC container.  I'll fork your branch and give the tests some love.

**Comment by bkuhl** [comment]
@brianwebb01 Are you ok with me making changes to this TruckerServiceProvider that make this a Laravel-only package?

**Comment by brianwebb01** [comment]
The world is your oyster. Do as you wish.

**Comment by bkuhl** [comment]
Digging deeper into this, there are a lot of dependencies that are no longer provided in some of Illuminate's package that this project relies on for the tests to pass.  I'm not going to spend the time to update this project to make the tests pass because the work is significant.  However I would like to make your branch available to others.

Are you able to change the MR for this branch so instead of going into `master` it goes into a `laravel-5.2-dev` branch?

**Comment by david-treblig** [comment]
I can't seem to figure out how to change it, but I'll try to initiate a new one instead.

**Comment by bkuhl** [comment]
Branch `laravel-5.2-dev` now exists

**Comment by bkuhl** [comment]
Changes merged in on https://github.com/Indatus/trucker/pull/27

**Pull request closed without merging**